### ISSUE TITLE
Will this work with Hero6?

### ISSUE THREAD

**Issue by garydelarosausername** [KonradIT/GoProStream#18]
I hope it will!

**Comment by KonradIT** [comment]
yes!

**Comment by garydelarosausername** [comment]
Nice!

**Comment by garydelarosausername** [comment]
Nope, it didn't work. Here's the traceback: https://pastebin.com/zncU2YeS

**Comment by KonradIT** [comment]
interesting, was the Hero6 up to date?

**Comment by garydelarosausername** [comment]
@KonradIT Yes.

**Comment by KonradIT** [comment]
Go to http://10.5.5.9/gp/gpControl in your browser, make sure the GoPro is connected to the computer, in Wifi App mode and is turned on. I won't get the Hero6 so you'll need to help me debug it.

**Comment by KonradIT** [comment]
copy the text that will appear and paste it into pastebin.com and share the link here.

**Comment by garydelarosausername** [comment]
Ok hold on...

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by KonradIT** [comment]
@garydelarosausername  Did you manage to get it?

**Comment by garydelarosausername** [comment]
Sorry I fell asleep :D I'm working on it right now.

**Comment by garydelarosausername** [comment]
https://pastebin.com/4wz5fbEt

**Comment by KonradIT** [comment]
Cool thanks!! Its strange that the script does not work, in theory it should. Let's test if the commands are the same: open this [http://10.5.5.9/gp/gpControl/execute?p1=gpStream&a1=proto_v2&c1=restart] on your browser with the HERO6 connected and turned on. It should return `{}`. If it does let me know and we'll continue debugging.

**Comment by garydelarosausername** [comment]
{"status":"0" }

That's what showed.

**Comment by KonradIT** [comment]
so it works, try running the updated script from this repo.

**Comment by garydelarosausername** [comment]
Ok. Thanks!

**Comment by garydelarosausername** [comment]
It's working! Holy crap! Thanks!

**Comment by garydelarosausername** [comment]
How do I project the exact same resolution to the laptop/PC? Like, 4k - especially. Right now, as you know too, it's only 720 that it shows on the PC.

**Comment by KonradIT** [comment]
4K is not supported lol, it would melt the camera. HERO6 Black has 5GHz WiFi so it should stream 720p at 4MBit/s without any problems, you can check by running these commands first:

http://10.5.5.9/gp/gpControl/setting/64/7

http://10.5.5.9/gp/gpControl/setting/62/4000000

I tried that configuration on my H5B and it had artifacts.

**Comment by garydelarosausername** [comment]
Ok I'll try that. The URL used to view media from H5 to PC doesn't work. Perhaps the directory is different now? I'm sure the camera is connected to my PC.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by KonradIT** [comment]
Try going to these:

http://10.5.5.9:8080/videos/DCIM/100GOPRO

http://10.5.5.9:8080/videos/DCIM/

http://10.5.5.9:8080/videos/

http://10.5.5.9:8080/DCIM

or do a nmap scan.

**Comment by garydelarosausername** [comment]
I think it connects but the video doesn't show. It shows me contents, folders, etc. (MP4's, etc.) like it does with H5.

**Comment by garydelarosausername** [comment]
Yes, I have an Android phone. LG K10 LTE, Android 6 version -said that to help with these awesome GoPro hacks :D

**Comment by KonradIT** [comment]
Cool, it would be nice to get the digital-zoom parameters. Install an app from the play store called Packet Capture and start logging (make sure you use VPN function) then open the GoPro app, connect to the H6B and zoom in and out (idk how that works yet, I think you just slide over the preview) and then go back to Packet Capture and stop logging. This is a bit tedious but look thru the GoPro app requests, screenshot the ones that have http://10.5.5.9/gp/gpControl/command/digital_zoom as the URL. this will be amazing. If not you can use Wireshark/tcpdump and install a MITM to get the URL but the app method is easy.

**Comment by garydelarosausername** [comment]
@KonradIT The videos doesn't show, audio plays though.

**Comment by KonradIT** [comment]
what do you mean the video doesn't show?

**Comment by garydelarosausername** [comment]
When I go to the URL in Chromium to view my videos from my camera, the video itself doesn't show but there's the audio playing.

**Comment by KonradIT** [comment]
Maybe it doesn't support H.265

**Comment by garydelarosausername** [comment]
By videos not showing but there's audio playing, I'm talking about http://10.5.5.9:8080/videos/DCIM/  and http://10.5.5.9:8080/videos/ They're the only ones that work too or that show stuff/files/contents.

**Comment by garydelarosausername** [comment]
Are you talking about the HVEC?

**Comment by garydelarosausername** [comment]
Damn it! Even previewing clips on the gallery in the camera doesn't work properly. It almost always freezes/lags for too long when I try to view the clips in the built-in LCD. Everything's updated and I'm even using the recommended Sandisk Extreme Plus 64gb by GoPro. The closest thing I can think of of me might causing problems is using my H5 batteries, which shouldn't be a problem because GoPro said it's compatible with H5 batteries.

**Comment by garydelarosausername** [comment]
@KonradIT I've thought of being able to zoom in and out while recording - hopefully this is a step towards that 😄 I'll look into this ASAP.

**Comment by rpowell028** [comment]
I am having an issue running this for the Hero 6. When I execute the python script I get this message back in cmd and that's it. No video feed opens with a livestream. I have ffmpeg installed. What could be going wrong?

HERO6 Black
HD6.01.00.26.00
UDP target IP: 10.5.5.9
UDP target port: 8554
message: _GPHD_:0:0:2:0.000000

Recording on camera: false

**Comment by rpowell028** [comment]
@KonradIT any ideas? Would love to get this working!

**Comment by KonradIT** [comment]
hi @rpowell028 what's your OS?

**Comment by rpowell028** [comment]
hi @KonradIT I'm on windows 10
